docs/layout-of-source-files.rst

```diff
@@ -35,6 +43,35 @@ the exact version of the compiler, so that bugfix releases are still possible.
 It is possible to specify much more complex rules for the compiler version,
 the expression follows those used by `npm <https://docs.npmjs.com/misc/semver>`_.
 
+.. index:: ! pragma, experimental
+
+.. _experimental_pragma:
+
+Experimental Pragma
+-------------------
+
+The second pragma is the experimental pragma. It can be used to enable
+features of the compiler or language that are not yet officially active.
+The following experimental pragmas are currently supported:
+
+
+ABIEncoderV2
+~~~~~~~~~~~~
+
+The new ABI encoder is able to encode and decode arbitrarily nested
+arrays and structs. It produces less optimal code (the optimizer
+for this part of the code is still under development) and has not
+received as much testing as the old encoder. You can activate it
+using ``pragma exprimental ABIEncoderV2;``.
+
+SMTChecker
+~~~~~~~~~~
+
+This component is not available in all Solidity binaries. If you use
```

Comment:
``This component has to be enabled at compile-time and is therefore not available in all Solidity binaries.``?